tools/release/README.md

```diff
@@ -19,27 +54,47 @@ Then, it's time to build the new beta version.
 
 ## Tag the release
 
+First you need to understand what tag to push. Checkbox uses
+[semantic versioning], in order to calculate what version number you should
+push, you need to see what you are actually releasing.
+
+Run the following commands to get a list of changes since the last stable
+release:
+
+```bash
+$ git fetch --tags
+$ git describe --tags --match "*[0123456789]" origin/main &&
+    git log $(
+        git describe --tags --match "*[0123456789]" --abbrev=0 origin/main
+    )..origin/main
+```
+
+If the list includes:
+- Only bugfixes: you can use the version tag that was calculated by the
+`edge-verification` job (increment patch).
+- New, backward compatible features: you have to increment the minor version
+- Non-backward compatible changes: you have to increment the major version
+
+If you were to be at at the tag `v2.9.1-edge-validation-success` and you
+had to release a new version with at least one backward compatible new feature,
+this is what that would look like:
+
 - Clone the repository
   ```
   git clone git@github.com:canonical/checkbox.git
   ```
 - Tag the release
   ```
-  git tag -s "v2.10.0" -m "Bump version: 2.9.1 → 2.10.0"
+  git tag -s "v2.10.0" -m "Beta promotion of: 2.9.1 → 2.10.0"
```

Comment:
For consistency with previous sample commands:
```suggestion
  $ git tag -s "v2.10.0" -m "Beta promotion of: 2.9.1 → 2.10.0"
```
This also applies to Line 84 and 92. (You can also remove all the prompt symbols :)